apps/server/src/repository/tx/transactionCollector2.ts

```diff
@@ -106,13 +106,15 @@ export class TransactionCollector2 {
       code_hash: Buffer.from(new Reader(script.codeHash).toArrayBuffer()),
       hash_type: script.hashType === 'data' ? 0 : 1,
     });
-    if ('0x' != script.args) {
+
+    if ('0x' != script.args && this.queryOptions.getLockArgsLen() !== -2) {
       query = query.whereRaw('substring(args, 1, ?) = ?', [argsBuffer.byteLength, argsBuffer]);
     }
 
-    // if (this.queryOptions.getArgsLen() !== 'any' && this.queryOptions.getArgsLen() > 0) {
-    //   query = query.whereRaw('length(args) = ?', [this.queryOptions.getArgsLen()]);
-    // }
+    if (script.args !== 'any' && this.queryOptions.getLockArgsLen() === -2 && type === 'lock') {
+      // query = query.whereRaw('substring(hex(args), 65, 128) = ?', [script.args.slice(2, 66)]);
+      query = query.whereRaw(`hex(args) like '%${script.args.slice(2, 66)}%'`);
```

Comment:
`like` will cause index to be unavailable

Reply:
This table affects okay